v2/cli/init_command.go

```diff
@@ -0,0 +1,256 @@
+		strings.ToLower(c.String(flagLanguage)),
+		c.String(flagGit),
+		"",
+	}
+
+	err := core.ValidateProjectID(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = file.EnsureDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate js/ts buffer
+	var scriptBytes []byte
+	scriptBytes, err = execTemplate(scriptTemplate, fields)
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	if fields.GitCloneURL != "" {
+		scriptPath := path.Join(".brigade", brigadeScriptFileName)
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+
+		// Also generate a minimal package.json file
+		var packageBytes []byte
+		packageBytes, err = execTemplate(packageTemplate, fields)
+		if err != nil {
+			return err
+		}
+
+		packagePath := path.Join(".brigade", "package.json")
+		if overwrite, err = checkFileOverwrite(packagePath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				packagePath,
+				packageBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Generate the project.yaml file
+	var projectBytes []byte
+	projectBytes, err = execTemplate(projectTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	projectPath := path.Join(".brigade", "project.yaml")
+	if overwrite, err = checkFileOverwrite(projectPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			projectPath,
+			projectBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var notesBytes []byte
+	notesBytes, err = execTemplate(notesTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	notesPath := path.Join(".brigade", "NOTES.txt")
+	if overwrite, err = checkFileOverwrite(notesPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			notesPath,
+			notesBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var secretsBytes []byte
+	secretsBytes, err = execTemplate(secretsTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	secretsPath := path.Join(".brigade", "secrets.yaml")
+	if overwrite, err = checkFileOverwrite(secretsPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			secretsPath,
+			secretsBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	err = addToGitignore(
+		"./.gitignore",
```

Comment:
This tripped me up initially. At a quick glance, it looks so much like `./.gitignore` is being added to `.gitignore`.

I know that's not what's actually happening. I know you've parameterized the file path _to make testing easier and that is a great move._

What can be done to make this easier to understand, though, is this:

Create _more_ generic function like `addLinesToFile(...)`. That will be the thing you can easily test.

`addToGitIgnore(...)` won't be tested, and it can invoke the well-tested `addLinesToFile(...)` function using the hard-coded `./.gitignore` path.

The end result is that when someone in the future looks at _this_ bit of code right here, they'll see:

```
err = addToGitIgnore(".brigade/secrets.yaml", ".brigade/node_modules/")
```

And that will be less likely to confuse them. They won't stop to ask themselves why `.gitignore` is being added to `.gitignore`.

Reply:
Should `addLinesToFile` be moved to the `file` package? Since it's now no longer specific to `init`

Reply:
That's a good question. imho, that's not a common enough operation that we should bother at this point. We can later if it crops up again.